avocado/core/runner.py

```diff
@@ -456,6 +456,9 @@ def sigtstp_handler(signum, frame):     # pylint: disable=W0613
             summary.add("INTERRUPTED")
         elif not mapping[test_state['status']]:
             summary.add("FAIL")
+            if getattr(test_factory[1]['job'].args, 'failfast', False):
+                self.job.log.debug("Interrupting job (failfast).")
```

Comment:
Since we're advertising that the job is being interrupted, we should be adding `AVOCADO_JOB_INTERRUPTED` to the job status.  And checking that in the functional tests is also a good idea.